src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -145,11 +147,17 @@ public override void SetResponse(JsonOperationContext context, BlittableJsonRead
         private long _operationId = -1;
         private string _nodeTag;
 
-        public CompressionLevel CompressionLevel = CompressionLevel.NoCompression;
         private readonly JsonSerializer _defaultSerializer;
         private readonly Func<object, IMetadataDictionary, StreamWriter, bool> _customEntitySerializer;
         private long _concurrentCheck;
 
+        public CompressionLevel CompressionLevel = CompressionLevel.NoCompression;
+
+        /// <summary>
+        /// Determines whether we should skip overwriting a document when it is updated by exactly the same document (by comparing the content and the metadata)
+        /// </summary>
+        public bool SkipOverwriteIfUnchanged;
```

Comment:
Can we re-introduce BulkInsertOptions, mark this readonly and add this there?

Reply:
done